assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py

```diff
@@ -319,3 +322,79 @@ def save_as_mlflow(self):
             conda_env=conda_env_file,
             code_path=code_path,
         )
+
+
+class LLaVAMLFlowConvertor(PyFuncMLFLowConvertor):
+    """PyFunc MLfLow convertor for LLaVA models."""
+
+    MODEL_DIR = os.path.join(os.path.dirname(__file__), "llava")
+
+    def __init__(self, **kwargs):
+        """Initialize MLflow convertor for LLaVA models."""
+        super().__init__(**kwargs)
+        if self._task != SupportedTasks.IMAGE_TEXT_TO_TEXT.value:
+            raise Exception("Unsupported task")
+
+    def get_model_signature(self) -> ModelSignature:
+        """Return MLflow model signature with input and output schema for the given input task.
+
+        :return: MLflow model signature.
+        :rtype: mlflow.models.signature.ModelSignature
+        """
+        input_schema = Schema(
+            [
+                ColSpec(DataType.string, LLaVAMLFlowSchemaLiterals.INPUT_COLUMN_IMAGE),
+                ColSpec(DataType.string, LLaVAMLFlowSchemaLiterals.INPUT_COLUMN_PROMPT),
+                ColSpec(DataType.string, LLaVAMLFlowSchemaLiterals.INPUT_COLUMN_DIRECT_QUESTION),
+            ]
+        )
+
+        output_schema = Schema(
+            [
+                ColSpec(DataType.string, LLaVAMLFlowSchemaLiterals.OUTPUT_COLUMN_RESPONSE)
+            ]
+        )
+
+        return ModelSignature(inputs=input_schema, outputs=output_schema)
+
+    def save_as_mlflow(self):
+        """Prepare model for save to MLflow."""
+        sys.path.append(self.MODEL_DIR)
+        from llava_mlflow_wrapper import LLaVAMLflowWrapper
+
+        mlflow_model_wrapper = LLaVAMLflowWrapper(task_type=self._task)
+        artifacts_dict = self._prepare_artifacts_dict()
+        conda_env_file = os.path.join(self.MODEL_DIR, "conda.yaml")
+        code_path = [
+            os.path.join(self.MODEL_DIR, "llava_mlflow_wrapper.py"),
+            os.path.join(self.MODEL_DIR, "config.py"),
+            os.path.join(self.MODEL_DIR, "utils.py")
+        ]
+        super()._save(
+            mlflow_model_wrapper=mlflow_model_wrapper,
+            artifacts_dict=artifacts_dict,
+            conda_env=conda_env_file,
+            code_path=code_path,
+        )
+
+    def _prepare_artifacts_dict(self) -> Dict:
+        """Prepare artifacts dict for MLflow model.
+
+        :return: artifacts dict
+        :rtype: Dict
+        """
+        # Get the name of the only subdirectory of the model directory.
+        sd = next(
+            (
+                d
```

Comment:
can we simplify this, since os.listdir does not return parent and current directory?

![image](https://github.com/Azure/azureml-assets/assets/61145377/9bab79a9-ab26-4c3c-be55-5210f5c9880f)

Reply:
Done and learned something in the process, thanks.